ou_dedetai/tui_app.py

```diff
@@ -23,109 +32,198 @@
-        self.confirm_q = Queue()
-        self.confirm_e = threading.Event()
-        self.password_q = Queue()
+        self.password_q: Queue[str] = Queue()
         self.password_e = threading.Event()
-        self.appimage_q = Queue()
+        self.appimage_q: Queue[str] = Queue()
         self.appimage_e = threading.Event()
-        self.install_icu_q = Queue()
-        self.install_icu_e = threading.Event()
-        self.install_logos_q = Queue()
-        self.install_logos_e = threading.Event()
+
+        self.terminal_margin = 0
+        self.resizing = False
+        # These two are updated in set_window_dimensions
+        self.console_log_lines = 0
+        self.options_per_page = 0
 
         # Window and Screen Management
-        self.tui_screens = []
-        self.menu_options = []
-        self.window_height = self.window_width = self.console = self.menu_screen = self.active_screen = None
-        self.main_window_ratio = self.main_window_ratio = self.menu_window_ratio = self.main_window_min = None
-        self.menu_window_min = self.main_window_height = self.menu_window_height = self.main_window = None
-        self.menu_window = self.resize_window = None
+        self.tui_screens: list[tui_screen.Screen] = []
+        self.menu_options: list[Any] = []
+
+        self._installer_thread: Optional[threading.Thread] = None
+
+        # Default height and width to something reasonable so these values are always
+        # ints, on each loop these values will be updated to their real values
+        self.window_height = self.window_width = 80
+        self.main_window_height = self.menu_window_height = 80
+        # Default to a value to allow for int type
+        self.main_window_min: int = 0
+        self.menu_window_min: int = 0
+
+        self.menu_window_ratio: Optional[float] = None
+        self.main_window_ratio: Optional[float] = None
+        self.main_window_ratio = None
+        self.main_window: Optional[curses.window] = None
+        self.menu_window: Optional[curses.window] = None
+        self.resize_window: Optional[curses.window] = None
+
+        # For menu dialogs.
+        # a new MenuDialog is created every loop, so we can't store it there.
+        self.current_option: int = 0
+        self.current_page: int = 0
+        self.total_pages: int = 0
+
+        # Start internal property variables, shouldn't be accessed directly, see their 
+        # corresponding @property functions
+        self._menu_screen: Optional[tui_screen.MenuScreen] = None
+        self._active_screen: Optional[tui_screen.Screen] = None
+        self._console: Optional[tui_screen.ConsoleScreen] = None
+        # End internal property values
+
+        # Lines for the on-screen console log
+        self.console_log: list[str] = []
+
+        # Note to reviewers:
+        # This does expose some possibly untested code paths
+        #
+        # Before some function calls didn't pass use_python_dialog falling back to False
```

Comment:
That was intentional, as the main menu I wanted to run in curses so that someone got the feedback which dialog couldn't give.